students/Steven/lesson05/mailroom03.py

```diff
@@ -0,0 +1,101 @@
+#! bin/user/env python3
+import sys
+from operator import itemgetter
+"""
+Program Part2:
+Update your mailroom program to:
+
+Use dicts where appropriate.
+See if you can use a dict to switch between the user’s selections.
+See if you can use a dict to switch between the users selections. see Using a Dictionary to switch for what this means.
+Convert your main donor data structure to be a dict.
+Try to use a dict and the .format() method to produce the letter as one big template,
+rather than building up a big string that produces the letter in parts.
+"""
+
+# Donor name, donation amount, donation history
+donor_list = {
+    "John Doe": [2000, 2],
+    "Jane Doe": [3000, 3],
+    "George Washington": [1, 1],
+    "Andrew Jackson": [20, 2],
+    "Benjamin Franklin": [100, 3]
+}
+
+# Main options menu for user to pick from
+def main():
+    while True:
+        options()  # Display each option to the user
+        try:
+            choice = int(input("What would you like to do? "))
+            # Using a dictionary for as a switch case
+            switch_func_dict = {
+                1: thank_you,
+                2: report,
+                3: all_letter,
+                4: quit
+            }
+            switch_func_dict.get(choice)()
+        except ValueError as error_message:
+            print(error_message)
+
+# Menu of options for the user in the script
+def options():
+    print("You have 4 options to choose from: ", "\n"
+        "1) Send Thank you", "\n"
+        "2) Create Report", "\n"
+        "3) Send a Thank you note to all donors", "\n"
+        "4) Exit", "\n"
+          )
+
+# Ask for a donor name and donation amount, send an email to that donor once completed
+def thank_you():
+    name = input("Enter a first and last name: ").title()
+    try:
+        if name.lower() == "list":
+            print(donor_list)
+            thank_you()
+        gift = float(input("Enter a donation amount: "))
+        if name in donor_list.keys():  # check if name is in the dict
+            donor_list[name][0] += gift  # Total balance of donations
+            donor_list[name][1] += 1  # Keeping track of donation history
+        else:
+            donor_list.update({name: [gift, 1]})
+    except ValueError as error_message:  # Catch if the user doesn't input a number
```

Comment:
you should have this around line 58 so it's clear what piece of code you're catching